source/vir/src/ast.rs

```diff
@@ -159,6 +159,11 @@ pub enum TypX {
     Lambda(Typs, Typ),
     /// Executable function types (with a requires and ensures)
     AnonymousClosure(Typs, Typ, usize),
+    /// Corresponds to Rust's FnDef type
+    /// Typs are generic type args
+    /// If Fun is a trait function, then the Option<Fun> has the statically resolved
+    /// function if it exists.
+    FnDef(Fun, Typs, ImplPaths, Option<Fun>),
```

Comment:
I didn't expect ImplPaths in TypX::FnDef, but I think it's actually not used.  Can it be removed?

Reply:
I put it there because TypX::Datatype has it as well (although I think that may also be unused?)

Reply:
The TypX::Datatype one is used in one place for checking datatype definitions (`recursive_types::check_positive_uses`, although it's not actually about positive uses).  Since TypX::FnDef can't appear in datatype definitions, I think that won't need it.